micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright 2020 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.kafka;
+
+import io.micrometer.core.annotation.Incubating;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNull;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+
+import static java.util.Collections.emptyList;
+
+/**
+ * Kafka metrics binder.
+ * <p>
+ * It is based on {@code metrics()} method returning {@link Metric} map exposed by clients and
+ * streams interface.
+ * <p>
+ * Meter names have the following convention: {@code kafka.(metric_group).(metric_name)}
+ *
+ * @author Jorge Quilcate
+ * @see <a href="https://docs.confluent.io/current/kafka/monitoring.html">Kakfa monitoring
+ * documentation</a>
+ * @since 1.4.0
+ */
+@Incubating(since = "1.4.0")
+@NonNullApi
+@NonNullFields
+class KafkaMetrics implements MeterBinder {
+    static final String METRIC_NAME_PREFIX = "kafka.";
+
+    static final String METRIC_GROUP_APP_INFO = "app-info";
+    static final String VERSION_METRIC_NAME = "version";
+    static final String START_TIME_METRIC_NAME = "start-time-ms";
+
+    private final Supplier<Map<MetricName, ? extends Metric>> metricsSupplier;
+    private final Iterable<Tag> extraTags;
+    private final int extraTagsSize;
+
+    /**
+     * Keeps track of current set of metrics. When this values change, metrics are bound again.
+     */
+    private volatile Map<MetricName, Meter> currentMeters = new HashMap<>();
+
+    private String kafkaVersion = "unknown";
```

Comment:
@shakuzen is this better than empty string?

Reply:
Yes, I think so.